### PR TITLE
feat(query): add collation

### LINKED ISSUE #8104: bug:  substr should consider string encodings
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

xx

### What's Wrong?

```select substr('城区-主城区-其他', 1, 6)```   got `城区` while we expect `城区-主城区`

### How to Reproduce?

select substr('城区-主城区-其他', 1, 6)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #8104

### CODE REVIEW

**Comment by sundy-li** [comment]
```
MySQL [(none)]> select substr('城区-主城区-其他', 1, 6), length('我爱中国');
+---------------------------------------------------+------------------------+
| substring('城区-主城区-其他' from 1 for 6)        | length('我爱中国')     |
+---------------------------------------------------+------------------------+
| 城区                                              |                     12 |
+---------------------------------------------------+------------------------+
1 row in set (0.030 sec)

MySQL [(none)]> set collation = 'utf8';
Query OK, 0 rows affected (0.028 sec)

MySQL [(none)]> select substr('城区-主城区-其他', 1, 6), length('我爱中国');
+---------------------------------------------------+------------------------+
| substring('城区-主城区-其他' from 1 for 6)        | length('我爱中国')     |
+---------------------------------------------------+------------------------+
| 城区-主城区                                       |                      4 |
+---------------------------------------------------+------------------------+
1 row in set (0.036 sec)
```

**Comment by BohuTANG** [comment]
CI checking failure :)

**Review by BohuTANG — APPROVED** [review]

File: src/query/settings/src/lib.rs [hunk 1]
```diff
@@ -364,6 +364,16 @@ impl Settings {
...
             },
+            SettingValue {
+                default_value: UserSettingValue::String("binary".to_owned()),
+                user_setting: UserSetting::create(
+                    "collation",
+                    UserSettingValue::String("binary".to_owned()),
+                ),
+                level: ScopeLevel::Session,
+                desc: "Char collation, support \"binary\" \"utf8\" default value: binary",
+                possible_values: Some(vec!["binary", "utf8"]),
```
Comment (sandflee) [inline src/query/settings/src/lib.rs:375]: why “binary” is the default value not utf-8？

Comment (sundy-li) [reply src/query/settings/src/lib.rs:375]: Because the string column is stored as binary by default.

Comment (sundy-li) [reply src/query/settings/src/lib.rs:375]: You can set global to make it default to use utf8

**Pull request merged**